src/jvm/test/org/jetbrains/kotlinx/lincheck_test/datastructures/LockFreeTaskQueue.kt

```diff
@@ -0,0 +1,322 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2025 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.jetbrains.kotlinx.lincheck_test.datastructures
+
+import kotlinx.atomicfu.*
+import kotlinx.coroutines.*
+import kotlin.jvm.*
+
+// Implementation take from https://github.com/Kotlin/kotlinx.coroutines/blob/1.7.3/kotlinx-coroutines-core/common/src/internal/LockFreeTaskQueue.kt
+private typealias Core<E> = LockFreeTaskQueueCore<E>
```

Comment:
Please move below, `LockFreeTaskQueue` should come first in the file (for better readability).